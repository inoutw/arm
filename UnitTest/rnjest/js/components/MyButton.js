import React, { Component } from 'react';
import { Button } from 'react-native';

export default class RnButton extends Component {
	state = {color: 'blue'}
	render() {
		return (
			<Button onPress={this._onPress} 
				onMouseEnter={this._onMouseEnter}
				onMouseLeave={this._onMouseLeave}
				title="submit" color={this.state.color}
			/>
		)
	}
	_onPress = () => {
		this.props.onPress();
	}
	_onMouseEnter = () => {
		this.setState({color: 'green'})
	}
	_onMouseLeave = () => {
		this.setState({ color: 'blue' })
	}
}