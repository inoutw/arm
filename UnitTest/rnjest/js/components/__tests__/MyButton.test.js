import React from 'react';
import renderer from 'react-test-renderer';
import MyButton from '../MyButton';

// https://github.com/facebook/jest/issues/1720#issuecomment-247746586
// jest.mock('Button', () => {
// 	const mockComponent = require('jest-react-native');
// 	return mockComponent('Button');
// });
jest.mock('Button', () => 'Button');

it('change button color', () => {
	const comp = renderer.create(
		<MyButton/>
	)
	let tree = comp.toJSON();
	// console.log('tree props', tree);
	expect(tree).toMatchSnapshot();

	// trigger the component method 
	tree.props.onMouseEnter();
	tree = comp.toJSON();
	expect(tree).toMatchSnapshot();

	tree.props.onMouseLeave();
	tree = comp.toJSON();
	expect(tree).toMatchSnapshot();	
})

it('should trigger the onPress handler', () => {
	let mockPress = jest.fn();
	const comp = renderer.create(
		<MyButton onPress={mockPress}/>
	)
	comp.toJSON().props.onPress();
	expect(mockPress).toBeCalled();
})