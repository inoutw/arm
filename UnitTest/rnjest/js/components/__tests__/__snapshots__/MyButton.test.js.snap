// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change button color 1`] = `
<Button
  color="blue"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onPress={[Function]}
  title="submit"
/>
`;

exports[`change button color 2`] = `
<Button
  color="green"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onPress={[Function]}
  title="submit"
/>
`;

exports[`change button color 3`] = `
<Button
  color="blue"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onPress={[Function]}
  title="submit"
/>
`;
